.eslintrc.js

```diff
@@ -114,14 +114,47 @@ module.exports = {
 			{
 				files: [ '**/*.ts', '**/*.tsx' ],
 				rules: {
+					// Disable vanilla eslint rules that have a Typescript implementation
+					// See https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/README.md#extension-rules
+					'brace-style': 'off',
+					'comma-dangle': 'off',
+					'comma-spacing': 'off',
+					'default-param-last': 'off',
+					'dot-notation': 'off',
+					'func-call-spacing': 'off',
+					indent: 'off',
+					'init-declarations': 'off',
+					'keyword-spacing': 'off',
+					'lines-between-class-members': 'off',
+					'no-array-constructor': 'off',
+					'no-dupe-class-members': 'off',
+					'no-duplicate-imports': 'off',
+					'no-empty-function': 'off',
+					'no-extra-parens': 'off',
+					'no-extra-semi': 'off',
+					'no-invalid-this': 'off',
+					'no-loop-func': 'off',
+					'no-loss-of-precision': 'off',
+					'no-magic-numbers': 'off',
+					'no-redeclare': 'off',
+					'no-shadow': 'off',
+					'no-unused-expressions': 'off',
+					'no-unused-vars': 'off',
+					'no-use-before-define': 'off',
+					'no-useless-constructor': 'off',
+					quotes: 'off',
+					'require-await': 'off',
+					'return-await': 'off',
+					semi: 'off',
+					'space-before-function-paren': 'off',
```

Comment:
Wow, seems that the default `eslint-recommended` is seriously incomplete 🙁

By the way, I don't think we use the `@typescript-eslint/eslint-plugin`s `eslint-recommended` config at this moment, so for 100% correctness, this list should include all rules from there, too. E.g., `no-dupe-args`.

Reply:
I added the rules from `@typescript-eslint/eslint-plugin`. The final set of errors is:

```
❯ @typescript-eslint/ban-ts-comment:                   3|
  @typescript-eslint/ban-ts-ignore:                    3|
  @typescript-eslint/ban-types:                      102|
  @typescript-eslint/explicit-module-boundary-types: 516|
  @typescript-eslint/no-empty-function:                2|
  @typescript-eslint/no-explicit-any:                 56|
  @typescript-eslint/no-inferrable-types:              2|
  @typescript-eslint/no-unused-vars:                   1|
  @typescript-eslint/no-use-before-define:            81|
  import/no-extraneous-dependencies:                   3|
  jsdoc/check-param-names:                             5|
  jsdoc/newline-after-description:                     3|
  jsdoc/require-param:                                13|
  jsdoc/require-property-description:                  2|
  no-restricted-imports:                               1|
  react-hooks/exhaustive-deps:                        13|
```